README

```diff
@@ -274,3 +274,101 @@ refresh			Attempts to reinsert all modules that are currently
+Therefore, the disabled mode is not recommended.
+In the **permissive** mode, SELinux is active, the security policy is loaded, the file system is labeled and access
+denial entries are logged. However, the policy is not enforced and thus no access is actually denied.
+In the **enforced** mode, the security policy is applied. Each access that is not explicitly allowed by the policy
+is denied.
+
+To be able to check SELinux for possible access denials, need to set the **permissive** mode in the **meta-aos-vm**
+for **refpolicy-aos** recipe to the **DEFAULT_ENFORCING** variable. This will make it possible to use SELinux,
+but in logging mode.
+
+To diagnose denials security policies, using the following command `journalctl | grep avc`, if the log contains
+**avc: denied** that means it is an SELinux policy denial. This log will include a line like this:
+
+```
+AVC avc:  denied  { write } for  pid=1248 comm="dnsmasq" name="dev-log" dev="tmpfs" ino=20945
+scontext=system_u:system_r:dnsmasq_t:s0 tcontext=system_u:object_r:syslogd_runtime_t:s0 tclass=sock_file permissive=0
+```
+
+- **scontext** - SELinux context of the process (source) that attempted the denied action.
+- **tcontext** -SELinux context of the object (target) the process attempted to access
+- **comm** - name of the command that was used to access the target context name
+- **tclass** - target object class
+- **dev** - device on which the target resides
+- **name** - name of the target
+- can contained **path** -  path to the object (target) the process attempted to access.
+
+The next step is to analyze the received log and find the corresponding **te** file by **scontext**, for example,
+for **dnsmasq_t** there is a corresponding **dnsmasq.te** file and in this file need to add a rule that
+allows **scontext** to access **tcontext**.
+
+### Recommended steps for editing refpolicy.
+
+To find the required **te** file where to add the rule need to find a line declaring the **scontext** type
+(label) how in example `type dnsmasq_t;`. Next, after the **Local policy comment**, add the following rule.
+Sometimes the rules are logically grouped, for example, there are rules for working with the `/etc` directory, etc.
+
+**It is important** that before adding a rule, need to look for a possible macro for **tcontext** declared in a
+file with the **if** extension, to find the necessary file **if**, first need to find the corresponding **te** file
+in which the label will be declared and then take the **if** file with the same name. For example, the
+`syslogd_runtime_t` label is declared in the `logging.te` file, then the macro should be looked for in the
+`logging.if` file.
+
+If some file or directory marked with an incorrect label or `unlabeled_t`, then  need to find the **te**
+file where the label of interest is declared and take the file with the **fc** extension and declare the context for
+the label of interest there. For example
+```
```

Comment:
Add empty line before and after code snippets. I suggest to install markdown linter for vscode.

Reply:
Done